mqtt/mqtt3/src/proto/packet.rs

```diff
@@ -159,16 +163,8 @@ impl PacketMeta for Connect {
         let protocol_name = super::Utf8StringDecoder::default()
             .decode(&mut src)?
             .ok_or(super::DecodeError::IncompletePacket)?;
-        if protocol_name != "MQTT" {
```

Comment:
As discussed, you should add this and the protocol level check back.

Reply:
True, but Mike build logic around these fields. Broker decides to disconnect a client, but not the connection itself. We can change this is the future. I decided to leave it like this for now.

Reply:
We can add these checks back in. We will also need to add the protocol_name and protocol_level to the connect packet.